classes/search/cmsfield.php

```diff
@@ -149,41 +89,6 @@ public function get_document($record, $options = []) {
         return $doc;
     }
 
-    /**
-     * Get default value for cms custom field.
-     *
-     * @return array
-     */
-    protected function get_default_values() {
-        global $DB;
-        if (is_null($this->defaultvalues)) {
-            $defaultvalues = [];
-            $sql = "SELECT mcf.id fieldid, mct.id typeid, mcf.configdata, mcf.name fieldname
-                      FROM {cms_types} mct
-                      JOIN {customfield_category} mcc ON mcc.itemid = mct.id
-                      JOIN {customfield_field} mcf ON mcf.categoryid = mcc.id
-                     WHERE mcc.component = 'mod_cms' AND mcc.area = 'cmsfield' AND mcf.type IN ('textarea', 'text')
-                  ORDER BY mct.id, mcf.sortorder";
-            $cmstypes = $DB->get_records_sql($sql);
-            foreach ($cmstypes as $cmstype) {
-                if (empty($defaultvalues[$cmstype->typeid])) {
-                    $data = new \stdClass();
-                    $configdata = json_decode($cmstype->configdata);
-                    $data->value = $configdata->defaultvalue ?? 'Default value';
-                    $data->valueformat = $configdata->defaultvalueformat ?? 0;
-                    $data->fieldname = $cmstype->fieldname;
-                } else {
-                    $data = $defaultvalues[$cmstype->typeid];
-                    $configdata = json_decode($cmstype->configdata);
-                    $data->value .= ', ' . $configdata->defaultvalue;
-                }
-                $defaultvalues[$cmstype->typeid] = $data;
-            }
-            $this->defaultvalues = $defaultvalues;
-        }
-        return $this->defaultvalues;
-    }
-
```

Comment:
Feels like we can remove following methods as they are implemented in the base class:

- check_access
- get_doc_url
- get_context_url

We won't need `get_data` method in this case

Reply:
Removed.